File: InPlan.Web/Modules/ServerTypes/Parametros/ProyectosTecnologiasForm.ts
import { IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface ProyectosTecnologiasForm {
    IdProyecto: IntegerEditor;
    IdTecnologia: IntegerEditor;
}

export class ProyectosTecnologiasForm extends PrefixedContext {
    static formKey = 'Parametros.ProyectosTecnologias';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ProyectosTecnologiasForm.init)  {
            ProyectosTecnologiasForm.init = true;

            var w0 = IntegerEditor;

            initFormType(ProyectosTecnologiasForm, [
                'IdProyecto', w0,
                'IdTecnologia', w0
            ]);
        }
    }
}

File: InPlan.Web/Modules/Parametros/ProyectosTecnologias/ProyectosTecnologiasForm.cs
using Serenity.ComponentModel;

namespace InPlan.Parametros.Forms;

[FormScript("Parametros.ProyectosTecnologias")]
[BasedOnRow(typeof(ProyectosTecnologiasRow), CheckNames = true)]
public class ProyectosTecnologiasForm
{
    public int IdProyecto { get; set; }
    public int IdTecnologia { get; set; }
}